lib/orbit/authentication/user.ex

```diff
@@ -38,4 +42,17 @@ defmodule Orbit.Authentication.User do
       name: :users_email_index
     )
   end
+
+  @spec get_names_from_employee(t()) :: map() | nil
+  def get_names_from_employee(struct) do
+    List.first(
+      Repo.all(
+        from(e in Employee,
+          where: e.email == ^struct.email,
+          select: %{first_name: e.first_name, preferred_first: e.preferred_first}
+        )
+      ),
+      nil
+    )
+  end
```

Comment:
**suggestion:** if we always plan on applying logic to the result to use the preferred name if present, should we just do that here? And if so, maybe call this something a bit more specific like `get_display_name_from_employee`?

Reply:
Also I think it would be good to somehow prevent the case of duplicate emails going forward, maybe in a separate ticket / PR.

Reply:
e7b8ea0

I agree re: duplicate 👍 I'll file a task